### ISSUE TITLE
FileDownload only becomes button after pn.widgets.Button is called

### ISSUE THREAD

**Issue by Hoxbro** [holoviz/panel#1920]
#### ALL software version info
(this library, plus any other relevant software, e.g. bokeh, python, notebook, OS, browser, etc)
* panel 0.11.0a6.post12+gd9e9aafa
* python 3.7.9
* Windows 10

#### Description of expected behavior and the observed behavior
FileDownload only becomes a button after `pn.widgets.Button` is called.

In a notebook it seems to "remember" that the `pn.widgets.Button` has been called. But if I clear the output and close and halt the notebook I need to run `pn.widgets.Button` again.

#### Complete, minimal, self-contained example code that reproduces the issue

```python
import panel as pn
from bokeh.sampledata.autompg import autompg

from io import StringIO

pn.extension()

def callback():
    sio = StringIO()
    autompg.to_csv(sio)
    sio.seek(0)
    return sio

pn.widgets.FileDownload(callback, embed=False, filename='autompg.csv', auto=True)

pn.widgets.Button()

```

#### Screenshots or screencasts of the bug in action

https://user-images.githubusercontent.com/19758978/104920376-f6c56f00-5997-11eb-9982-f85a8bcf76d2.mp4

**Comment by philippjfr** [comment]
Interesting, can you check if this still occurs if you build against https://github.com/holoviz/panel/pull/1889?

**Comment by Hoxbro** [comment]
I don't know if I doing something wrong but now I get no buttons...

![2021-01-18 15_39_22](https://user-images.githubusercontent.com/19758978/104929507-46119c80-59a4-11eb-88b7-284d46e93fcc.png)


<details>
  <summary>Console log</summary>

```
This page uses the non standard property “zoom”. Consider using calc() in the relevant property values, or using “transform” along with “transform-origin: 0 0”. download.ipynb
Failed to get subsystem status for purpose Error: Could not establish connection. Receiving end does not exist.
    sendMessage moz-extension://e3de4889-f5a3-468c-8e62-98ce48465897/content-utils.js:47
content-script.js:95:21
Use of Mutation Events is deprecated. Use MutationObserver instead. jquery.min.js:2
actions jupyter-notebook:find-and-replace does not exist, still binding it in case it will be defined later... menubar.js:282
accessing "actions" on the global IPython/Jupyter is not recommended. Pass it to your objects constructors at creation time main.js:216
Loaded moment locale en bidi.js:19
load_extensions 
Arguments { 0: "jupyter-matplotlib/extension", 1: "plotlywidget/extension", 2: "jupyter-js-widgets/extension", … }
utils.js:60
Loading extension: jupyter-matplotlib/extension utils.js:37
Loading extension: plotlywidget/extension utils.js:37
Bokeh: BokehJS not loaded, scheduling load and callback at 
Date Mon Jan 18 2021 15:39:06 GMT+0100 (Central European Standard Time)
main.min.js:44:13
Bokeh: injecting link tag for BokehJS stylesheet:  https://unpkg.com/tabulator-tables@4.9.3/dist/css/tabulator_simple.min.css main.min.js:67:15
Bokeh: injecting script tag for BokehJS library:  https://unpkg.com/tabulator-tables@4.9.3/dist/js/tabulator.js main.min.js:86:15
Bokeh: injecting script tag for BokehJS library:  https://unpkg.com/moment@2.27.0/moment.js main.min.js:86:15
Loading extension: jupyter-js-widgets/extension utils.js:37
Uncaught Error: Mismatched anonymous define() module: function () {

	'use strict';

	// https://tc39.github.io/ecma262/#sec-array.prototype.findIndex


	if (!Array.prototype.findIndex) {

		Object.defineProperty(Array.prototype, 'findIndex', {

			value: function value(predicate) {

				// 1. Let O be ? ToObject(this value).


				if (this == null) {

					throw new TypeError('"this" is null or not defined');
				}

				var o = Object(this);

				// 2. Let len be ? ToLength(? Get(O, "length")).


				var len = o.length >>> 0;

				// 3. If IsCallable(predicate) is false, throw a TypeError exception.


				if (typeof predicate !== 'function') {

					throw new TypeError('predicate must be a function');
				}

				// 4. If thisArg was supplied, let T be thisArg; else let T be undefined.


				var thisArg = arguments[1];

				// 5. Let k be 0.


				var k = 0;

				// 6. Repeat, while k < len


				while (k < len) {

					// a. Let Pk be ! ToString(k).


					// b. Let kValue be ? Get(O, Pk).


					// …
    makeError http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:168
    intakeDefines http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:1251
    localRequire http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:1449
require.js:168:17
Uncaught Error: Mismatched anonymous define() module: function () { 'use strict';

    var hookCallback;

    function hooks() {
        return hookCallback.apply(null, arguments);
    }

    // This is done to register the method called with moment()
    // without creating circular dependencies.
    function setHookCallback(callback) {
        hookCallback = callback;
    }

    function isArray(input) {
        return (
            input instanceof Array ||
            Object.prototype.toString.call(input) === '[object Array]'
        );
    }

    function isObject(input) {
        // IE8 will treat undefined and null as object if it wasn't for
        // input != null
        return (
            input != null &&
            Object.prototype.toString.call(input) === '[object Object]'
        );
    }

    function hasOwnProp(a, b) {
        return Object.prototype.hasOwnProperty.call(a, b);
    }

    function isObjectEmpty(obj) {
        if (Object.getOwnPropertyNames) {
            return Objec…
    makeError http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:168
    intakeDefines http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:1251
    localRequire http://localhost:8888/static/components/requirejs/require.js?v=d37b48bb2137faa0ab98157e240c084dd5b1b5e74911723aa1d1f04c928c2a03dedf922d049e4815f7e5a369faa2e6b6a1000aae958b7953b5cc60411154f593:1449
require.js:168:17
Bokeh: all BokehJS libraries/stylesheets loaded main.min.js:50:17
Bokeh: BokehJS plotting callback run at 
Date Mon Jan 18 2021 15:39:07 GMT+0100 (Central European Standard Time)
main.min.js:1121:15
[bokeh] setting log level to: 'info' main.min.js:307:1356
Uncaught Error: Cannot find module './custom_html'
    require http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:212
    _ http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1022
    require http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:230
    _ http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1021
    require http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:230
    register_plugin http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:270
    inline_js http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1015
    inline_js http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1020
    inline_js http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1009
    inline_js http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1010
    run_inline_js http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1107
    <anonymous> http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:1122
    run_callbacks http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:23
    run_callbacks http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:21
    on_load http://localhost:8888/static/notebook/js/main.min.js?v=3d694ae0395dfef075eb0c5e8f10a21a8689f66b1e109f0c303b3573fd6512da3aebdc33284fe88676520a882517a504aec0051332d7d513d7c542465fc96b6f line 35619 > eval:51
main.min.js:212:27
Session: kernel_created (bb93f83a-16da-4f86-bcf8-2a555e8fd8f7) session.js:54
Starting WebSockets: ws://localhost:8888/api/kernels/552e5223-97ff-4cee-9a84-bb2e375ab8d5 kernel.js:463
Kernel: kernel_connected (552e5223-97ff-4cee-9a84-bb2e375ab8d5) kernel.js:106
Bokeh: BokehJS not loaded, scheduling load and callback at 
Date Mon Jan 18 2021 15:39:10 GMT+0100 (Central European Standard Time)
main.min.js:44:13
Bokeh: injecting link tag for BokehJS stylesheet:  https://unpkg.com/tabulator-tables@4.9.3/dist/css/tabulator_simple.min.css main.min.js:67:15
Bokeh: injecting script tag for BokehJS library:  https://unpkg.com/tabulator-tables@4.9.3/dist/js/tabulator.js main.min.js:86:15
Bokeh: injecting script tag for BokehJS library:  https://unpkg.com/moment@2.27.0/moment.js main.min.js:86:15
Kernel: kernel_ready (552e5223-97ff-4cee-9a84-bb2e375ab8d5) kernel.js:106
Bokeh: all BokehJS libraries/stylesheets loaded main.min.js:50:17
Bokeh: BokehJS plotting callback run at 
Date Mon Jan 18 2021 15:39:10 GMT+0100 (Central European Standard Time)
main.min.js:1121:15
[bokeh] setting log level to: 'info' main.min.js:307:1356
Bokeh: all callbacks have finished main.min.js:28:13
Bokeh: ERROR: Unable to run BokehJS code because BokehJS library is missing 2 download.ipynb line 138 > injectedScript:19:19
```
</details>

**Comment by philippjfr** [comment]
Did you recompile with `panel build panel`?

**Comment by philippjfr** [comment]
The steps to reproduce:

1. `panel build panel`
2. Clear notebook
3. Reload notebook page
4. Rerun

**Comment by Hoxbro** [comment]
When I run `panel build panel` I get the following error message:
![2021-01-18 16_05_59](https://user-images.githubusercontent.com/19758978/104931896-55461980-59a7-11eb-9edc-12a8876f566a.png)

Which is the same problem I saw in my console log. 

I also tried to update `"@bokeh/bokehjs": "^2.3.0-dev.10"` in `package.json` still no success.

**Comment by philippjfr** [comment]
You can ignore those for now.

**Comment by Hoxbro** [comment]
I still can't get the buttons to show up and I get the same error as before. I have observed that the console log give error `Uncaught Error: Cannot find module './custom_html'` which was also present when running `panel build panel`.

To be fair I feel like I'm being held hostage in a conda environment purgatory, so there is a clear possibility that I have done something wrong...

**Comment by philippjfr** [comment]
No sorry, I think it's my fault, will have to clean up the PR.

**Comment by philippjfr** [comment]
Pushed a fix, sorry again.

**Comment by Hoxbro** [comment]
Got the example running on bokeh 2.3, but I still see the problem: 
![image](https://user-images.githubusercontent.com/19758978/104951195-5e90af80-59c2-11eb-9415-08ff6e5d4c32.png)

**Comment by philippjfr** [comment]
Thanks for doing that. Hopefully that'll be fixed once Bokeh 2.3.0dev11 is out.

**Comment by Hoxbro** [comment]
Still seeing the problem on Bokeh 2.3.0dev11 and latest panel master:
![image](https://user-images.githubusercontent.com/19758978/105598515-8d2cd280-5d94-11eb-862e-d357b7450c1e.png)
